manager/integration/tests/test_basic.py

```diff
@@ -5031,23 +5032,45 @@ def test_delete_backup_during_restoring_volume():  # NOQA
     The volume robustness should be faulted if the backup was deleted during
     restoring the volume.
 
-    Setup:
+    1. Given create volume v1 and attach to a node
+       And write data 150M to volume v1
+    2. When create a backup of volume v1
+       And wait for that backup is completed
+       And restore a volume v2 from volume v1 backup
+       And delete the backup immediately
+    3. Then volume v2 "robustness" should be "faulted"
+       And "status" of volume restore condition should be "False",
+       And "reason" of volume restore condition should be "RestoreFailure"
+    """
+    # Step 1
+    vol_v1_name = "vol-v1"
+    vol_v1 = create_and_check_volume(client, vol_v1_name, size=str(512 * Mi))
+    vol_v1.attach(hostId=get_self_host_id())
+    vol_v1 = wait_for_volume_healthy(client, vol_v1_name)
 
-    1. Create a volume v1 and attach to a node
-    2. Write some data (>100M) to volume v1
+    vol_v1_endpoint = get_volume_endpoint(vol_v1)
+    write_volume_dev_random_mb_data(vol_v1_endpoint, 1, 150)
 
-    Steps:
+    # Step 2
+    bv, b, snap2, data = create_backup(client, vol_v1_name)
 
-    1. Create a backup of volume v1
-    2. Wait for that backup is completed
-    3. Start to restore a volume v2 from this backup
-       (Not wait for restoration completed)
-    4. Delete the backup immediately
-    5. Check that volume v2 "robustness" is "faulted" and
-       the status of volume restore condition is "False",
-       the reason of volume restore condition is "RestoreFailure"
-    """
-    pass
+    vol_v2_name = "vol-v2"
+    client.create_volume(name=vol_v2_name,
+                         size=str(512 * Mi),
+                         numberOfReplicas=3,
+                         fromBackup=b.url)
+
```

Comment:
> LGTM
> 
> This case (Deleting the backup immediately after restoring of volume triggered) looks fine but I have another case also in my mind as below. In the implemented case, I believe by the time volume starts restore, the backup is deleted.
> 
> 1. How do we make sure that the volume is in middle of the restoring when the backup was deleted?

Hi @khushboo-rancher :
    I implemented the test case base on https://github.com/longhorn/longhorn/issues/1867#issuecomment-1325880709, but **Scenario 2** couldn't be reproduced. That's why, I don't add `wait_for_volume_restoration_start` before `delete_backup`. If we need to make sure that the volume is in the middle of the restoring when the backup was deleted, I need to create another test case for it.
**WDYT**  ?  
    
    
    
Scenario 1:
  - [X] Restore volume from backup then immediately delete the backup - volume become faulted immediately
  
Scenario 2:
  - [ ] Restore volume from backup then delete the backup while restoring was in progress - volume become faulted after awhile

Reply:
@roger-ryao could you confirm that we could delete the backup that is used to be the source data of the restoration and restoring is in progress? I could not delete the backup in master-head environment.

Reply:
> @roger-ryao could you confirm that we could delete the backup that is used to be the source data of the restoration and restoring is in progress? I could not delete the backup in master-head environment.

Hi @mantissahz :
     We have two scenarios. One is **Deleting the backup immediately after restoring of volume triggered**, we could delete the backup. This [PR-1249](https://github.com/longhorn/longhorn-tests/pull/1249) would cover it.
     Another one is **Deleting the backup  after restoring is in progress**, we couldn't delete the backup. 
     `test_basic.py::test_backup_lock_deletion_during_restoration`  have covered this scenario.

Reply:
@roger-ryao 

so Scenario 2:

>Restore volume from backup then delete the backup while restoring was in progress - volume become faulted after awhile

It will not happen, right?

Reply:
I can not reproduce scenario 2 in master-head and v1.4.x-head, maybe my test environment was  not stable when I was testing.

Reply:
> It will not happen, right

@mantissahz 
   No, scenario 2 won't happen. We should refer to `test_basic.py::test_backup_lock_deletion_during_restoration`  when testing scenario 2.